File: OpenCv-Cpp/FileOperations/FileOperations.cpp
#include <iostream>
#include <string>
#include <opencv2/core.hpp>
#include <opencv2/highgui.hpp>
#include <opencv2/imgproc.hpp>

int main()
{
	std::string path;
	std::cout << "Give Path for Image: ";
	std::cin >> path;
	// Load Image from path to a Matrix
	cv::Mat image;
	image = cv::imread(path);

	// make sure data is loaded
	if (!image.data)
	{
		std::cout << "\nImage File data not loaded.\nExiting...";
		return -1;
	}

	// create empty matrix for storing resized image
	cv::Mat resized;

	// create point with two variables for size
	cv::Point size = cv::Point(512, 512);
	cv::resize(image, resized, size);

	// Write Image matrix to Image file
	cv::imwrite("resizedOnOpencv.jpg", resized);

	// Display Images in GUI.
	cv::imshow("image", image);
	cv::imshow("imageResized", resized);

	// wait with delay for any key from GUI.
	cv::waitKey(0);
	cv::destroyAllWindows();
	return 0;
}

File: OpenCv-Python/FileOperations/FileOperations.py
#! python
import cv2

path = input("Enter path for Image:")

# Read image from path File
image = cv2.imread(path)

# check if image is loaded
if image is None:
    print('Image File data not loaded.\nExiting...')
    exit()

# Resize the image with width, height
width = 512
height = 512
resized = cv2.resize(image, (width, height))

# Write the image to file
cv2.imwrite("output/resizedOnOpencv.jpg", resized)

# show the image in a GUI
cv2.imshow('original', image)
cv2.imshow('resized', resized)

# after delay wait for any key from GUI
cv2.waitKey(0)

# destroy or free any remaining GUI functionality
cv2.destroyAllWindows()
